src/kernel/src/hal/platform/hyperlight/peb.rs

```diff
@@ -77,6 +77,16 @@ impl ProcessEnvironmentBlock {
         Ok(())
     }
 
+    pub unsafe fn get_credits() -> Result<u64, Error> {
+        match GUEST_HANDLE.peb() {
+            Some(peb_ptr) => Ok((*peb_ptr).credits_value),
+            None => {
+                error!("peb not initialized");
+                Err(Error::new(ErrorCode::NoSuchDevice, "peb not initialized"))
```

Comment:
Could you create a string with the reason of failure and use it to avoid string duplication? Also, could you tag the function name in the error message to make it machine-parsable?

```rust
let reason: &'static str= "peb not initialized";
error("get_credits(): {reason}");
Err(Error::new(ErrorCode::NoSuchDevice, reason))
```